### ISSUE TITLE
Cannot edit existing link in URLInput

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3350]
When inserting a link or a "@" username mention, clicking on a link or moving the caret on a link makes an inline "popup" appear. The popup contains controls to edit the link, remove the link, additional settings (e.g. open in a new window):

![screen shot 2017-11-06 at 15 23 54](https://user-images.githubusercontent.com/1682452/32447039-216d9ccc-c30b-11e7-9b1e-c3649f61a7de.png)

Although the popup is "inline" and it is possible to press Tab and move focus to it, there are a few missing features compared to the similar link inline toolbar in the Classic Editor:

- `Alt+F8` should move focus to the popup
- `Escape` should move focus back from the popup to the link, to the previous caret position or selection
- pressing Tab and arrow keys should loop through the popup controls
- tabbing should be constrained within the popup

Also, to match the current functionality in the Classic Editor, `Cmd/Ctrl + K` should switch the popup to edit mode. Currently, it is only possible to click the two edit buttons, with different results:

![screen shot 2017-11-06 at 16 05 50](https://user-images.githubusercontent.com/1682452/32447536-b0ec8e7a-c30c-11e7-9c95-5a90254ec74e.png)

- clicking the Edit button within the modal, will switch to edit mode populating the input field with the current link URL
- clicking the Edit button in the top fixed toolbar, will switch to edit mode but the input field is empty

Also, when editing an username "mention", what should happen? Should the popup appear in edit mode and give users the ability to manually edit the link to the user page or present again the list of username suggestions?

Current behavior:

![screen shot 2017-11-06 at 16 09 51](https://user-images.githubusercontent.com/1682452/32447736-2f6ef51c-c30d-11e7-804f-079441da56ff.png)

**Comment by JeroenSormani** [comment]
It may be an new issue / working before, but I believe also a accessibility case would be that the word that was selected to be linked is no longer highlighted;
![image](https://user-images.githubusercontent.com/5774447/38777688-2f804098-40ac-11e8-8102-1455b730b098.png)
The 'Hello' word is selected here to be linked.
(The positioning error is reported at https://github.com/WordPress/gutenberg/issues/6183)

**Comment by afercia** [comment]
Some time has passed since Nov 6 2017, when this issue was created and some things have changed. for example. mentions don't insert a link any longer.

However, it would be greatly beneficial to review again all the "insert link" popups, check the interaction consistency, and the points mentioned above.

**Comment by designsimply** [comment]
Agree about needing to re-check the interaction consistency. Added a `Needs Testing` label for that purpose.

<strong>Note:</strong> a separate issue has been filed at https://github.com/WordPress/gutenberg/issues/6156 for the issue @JeroenSormani mentioned where the link text does not appear to be selected when you are adding a link.

**Comment by tofumatt** [comment]
This will be fixed by #10983. 🎉
